Maple2.Tools/VectorMath/Transform.cs

```diff
@@ -89,9 +89,9 @@ public float Scale {
             float scale = Scale;
 
             // Normalize each axis before scaling to prevent floating point drift.
-            RightAxis = Vector3.Normalize(RightAxis) * (value / scale);
-            UpAxis = Vector3.Normalize(UpAxis) * (value / scale);
-            FrontAxis = Vector3.Normalize(FrontAxis) * (value / scale);
+            RightAxis = Vector3.Normalize(RightAxis) * value;
+            UpAxis = Vector3.Normalize(UpAxis) * value;
+            FrontAxis = Vector3.Normalize(FrontAxis) * value;
```

Comment:
What are you even talking about? The normalization is still there. It was using the original scale that was removed because the original scale is no longer a factor after normalization. That if check you recommended doesn't even do anything mathematically meaningful. It's a junk operation